library/src/main/java/com/novoda/downloadmanager/MigrationJob.java

```diff
@@ -53,6 +56,86 @@ public void run() {
         onUpdate(migrationStatus);
 
         Log.d(TAG, "about to extract migrations, time is " + System.nanoTime());
+        Log.d(TAG, "partial migrations are all EXTRACTED, time is " + System.nanoTime());
+
+        migratePartialDownloads(database, partialDownloadMigrationExtractor, downloadsPersistence);
+        migrateCompleteDownloads(migrationStatus, database, migrationExtractor, downloadsPersistence, internalFilePersistence);
+    }
+
+    private void onUpdate(MigrationStatus migrationStatus) {
+        for (MigrationCallback migrationCallback : migrationCallbacks) {
+            migrationCallback.onUpdate(migrationStatus);
+        }
+    }
+
+    private void migratePartialDownloads(SqlDatabaseWrapper database, PartialDownloadMigrationExtractor partialDownloadMigrationExtractor, DownloadsPersistence downloadsPersistence) {
+        List<Migration> partialMigrations = partialDownloadMigrationExtractor.extractMigrations();
+        for (int i = 0; i < partialMigrations.size(); i++) {
```

Comment:
Could use a For Each here instead I think 👍

Reply:
Ah true, I'm not using the index like in the other one.